### PR TITLE
Correct default dims keywords to match Base

### LINKED ISSUE #45: sort doesn't work
```julia
julia> a = [2, 5, 4];

julia> nda = NamedDimsArray(a, :a);

julia> sort(nda)
ERROR: MethodError: no method matching sort!(::Array{Int64,1}; dims=Colon())
```

### LINKED ISSUE #46: Base functions need different `dims` fallbacks
When extending Base functions that take a `dims` keyword [we set the default `dims=:`](https://github.com/invenia/NamedDims.jl/blob/45a358b6a9157bad5a5c683a2e2716d30614c2e6/src/functions.jl#L28), but this does not match Base in all cases. 

These are the cases we get right:

- `sum, prod, maximum, minimum, extrema` 
```julia
julia> using NamedDims, Test

julia> a = rand(3, 3);

julia> nda = NamedDimsArray{(:x, :y)}(a);

julia> for f in (sum, prod, maximum, minimum, extrema)
           @test f(a) == f(nda) == f(a; dims=:)
       end
```

- `mapreduce`
```julia
julia> a = rand(Int, 3, 3);

julia> nda = NamedDimsArray{(:x, :y)}(a);

julia> @test mapreduce(isodd, *, a) == mapreduce(isodd, *, a) == mapreduce(isodd, *, a; dims=:)
```

And these are the cases we need to fix:
- `count` does not take a `dims` keyword
```julia
julia> a = [true false; true true];

julia> count(a)
3

julia> count(a; dims=1)
ERROR: MethodError: no method matching count(::Array{Bool,2}; dims=1)

julia> nda = NamedDimsArray{(:x, :y)}(a);

julia> count(nda)
ERROR: MethodError: no method matching count(::Array{Bool,2}; dims=Colon())
```
- `sort`/`sort!` for multidimensional array does not have a default for the `dims` keyword, and the input must be a `Integer` (#45)
- `cumsum` and `cumprod` are the same as `sort`:  no default, must be `Integer`
-  `mapslices` does not have not have a default for the `dims` keyword, and the input must be a `Integer` OR an iterable of `Integer`s

```julia
julia> a = reshape(Vector(1:8),(2,2,2));

julia> nda = NamedDimsArray{(:x, :y, :z)}(a);

julia> mapslices(sum, a)
ERROR: UndefKeywordError: keyword argument dims not assigned
Stacktrace:
 [1] mapslices(::Function, ::Array{Int64,3}) at ./abstractarray.jl:1919
 [2] top-level scope at none:0

julia> mapslices(sum, a; dims=1)
1×2×2 Array{Int64,3}:
[:, :, 1] =
 3  7

[:, :, 2] =
 11  15

julia> mapslices(sum, a; dims=[1, 3])
1×2×1 Array{Int64,3}:
[:, :, 1] =
 14  22

julia> mapslices(sum, nda)
ERROR: MethodError: no method matching iterate(::Colon)
```

### PR DESCRIPTION
**Pull request by nickrobinson251**
- Closes #45
- Closes #46

### CODE REVIEW

File: src/functions.jl [hunk 1]
```diff
@@ -12,16 +12,12 @@ function nameddimsarray_result(original_nda, reduced_data, reduction_dims::Colon
     return reduced_data
 end
 
-
 ###################################################################################
 # Overloads
 
 # 1 Arg
```
Comment (oxinabox) [inline src/functions.jl]:
```suggestion
# 1 Arg: reducers 
```

Comment (oxinabox) [reply src/functions.jl]: Also rename `nameddimsarray_result` to `nameddimsarray_reduction_result` ?

File: src/functions.jl [hunk 2]
```diff
@@ -33,12 +29,31 @@ for (mod, funs) in (
     end
 end
 
-# 1 arg before
+# 1 Arg - no default for `dims` keyword
+for (mod, funs) in (
```
Comment (nicoleepp) [inline src/functions.jl]: For code clarity, might be better to change `funs` and `fun` to `funcs` and `func` which are the more common abbreviation of `functions`? Even if it does makes the code less "fun" :)

File: src/functions.jl [hunk 3]
```diff
@@ -33,12 +29,31 @@ for (mod, funs) in (
...
+    end
+end
+
+# 1 arg before - no default for `dims` keyword
 for (mod, funs) in (
     (:Base, (:mapslices,)),
 )
     for fun in funs
-        @eval function $mod.$fun(f, a::NamedDimsArray; dims=:, kwargs...)
+        @eval function $mod.$fun(f, a::NamedDimsArray; dims, kwargs...)
```
Comment (nicoleepp) [inline src/functions.jl]: How does this work without a default for the kwarg `dims`?

File: test/functions.jl [hunk 4]
```diff
@@ -3,43 +3,86 @@ using NamedDims: names
 using Test
 using Statistics
 
-@testset "sum" begin
-    nda = NamedDimsArray([10 20; 30 40], (:x, :y))
+@testset "Base" begin
+a = [10 20; 31 40]
```
Comment (nicoleepp) [inline test/functions.jl]: Code within the testset should be indented

File: src/functions.jl (hunk 3 above)
Comment (nickrobinson251) [reply src/functions.jl]: just the same as Base -- without `dims` specified by the user, it is a `UndefKeywordError`

```julia
julia> A = reshape(1:8, (2, 2, 2));

julia> mapslices(sum, A)
ERROR: UndefKeywordError: keyword argument dims not assigned
Stacktrace:
 [1] mapslices(::Function, ::Array{Int64,3}) at ./abstractarray.jl:1919
 [2] top-level scope at none:0
```

File: test/functions.jl (hunk 4 above)
Comment (nickrobinson251) [reply test/functions.jl]: I agree in principle, the problem is that this file was already way over the line length, and indenting it further made writing the code on my little laptop screen almost impossible

I think a better solution is a follow-up PR that fixes up the style in the test - what do you think?

Comment (nicoleepp) [reply test/functions.jl]: That seems good to me

File: src/functions.jl (hunk 2 above)
Comment (nickrobinson251) [reply src/functions.jl]: haha 🎊 

For this PR i would prefer to stick with these names as they're already used elsewhere in the file

File: test/functions.jl (hunk 4 above)
Comment (nickrobinson251) [reply test/functions.jl]: Cool - i'll merge this and make a follow up PR for style changes :) thanks!

**Pull request merged**